tools/find-unused-packages.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+USED=$(opam list -t --required-by xs-toolstack --recursive --short | sort)
+UPSTREAM=$(find "packages/upstream/" -mindepth 1 -maxdepth 1 -type d -print0 | xargs -0 -I {} echo "{}" | cut -d "/" -f 3 | cut -d "." -f 1 | sort)
+XS=$(find "packages/xs/" -mindepth 1 -maxdepth 1 -type d -print0 | xargs -0 -I {} echo "{}" | cut -d "/" -f 3 | cut -d "." -f 1 | sort)
+diff -u .github/unused-upstream.txt <(comm -23 <(echo "$UPSTREAM") <(echo "$USED"))
+diff -u .github/unused-xs.txt <(comm -23 <(echo "$XS") <(echo "$USED"))
```

Comment:
It'd be nice to print the diff to fix (set -e does not make this easy, silly bash)

```suggestion
UPSTREAM_DIFF=$(diff -u .github/unused-upstream.txt <(comm -23 <(echo "$UPSTREAM") <(echo "$USED")) || true)
XS_DIFF=$(diff -u .github/unused-xs.txt <(comm -23 <(echo "$XS") <(echo "$USED")) || true)

if [ -n "$XS_DIFF" ] || [ -n "$UPSTREAM_DIFF" ]; then
  echo "$XS_DIFF" 1>&2 ;
  echo "$UPSTREAM_DIFF" 1>&2 ;
  echo "ERROR please fix above diffs" 2>&2 ;
  exit 1
fi
```

Reply:
the diff is being printed, that's how I iterated on the changes (and saw that the checks were being done too soon)

example: https://github.com/xapi-project/xs-opam/runs/2631749150?check_suite_focus=true

Reply:
who doesn't?